backend/backend.go

```diff
@@ -822,17 +816,18 @@ func (backend *Backend) uninitAccounts() {
 	backend.accounts = []accounts.Interface{}
 }
 
-// Keystores returns the keystores registered at this backend.
-func (backend *Backend) Keystores() *keystore.Keystores {
-	return backend.keystores
+// Keystore returns the keystore registered at this backend, or nil if no keystore is registered.
+func (backend *Backend) Keystore() keystore.Keystore {
+	defer backend.accountsAndKeystoreLock.RLock()()
+	return backend.keystore
 }
 
 // registerKeystore registers the given keystore at this backend.
 func (backend *Backend) registerKeystore(keystore keystore.Keystore) {
+	defer backend.accountsAndKeystoreLock.Lock()()
+
 	backend.log.Info("registering keystore")
-	if err := backend.keystores.Add(keystore); err != nil {
-		backend.log.Panic("Failed to add a keystore.", err)
-	}
+	backend.keystore = keystore
```

Comment:
Just checking: what if backend.keystore is not nil, where does the "old" keystore go? Shouldn't you check and return an error? oh wait... there's no returned error :)

Or it doesn't matter? I know you might say this can't happen but still... maybe at least log a message?

Reply:
It will simply be replaced. In devmode, this happens for example if you register a software keystore and then unlock a bb02.

I added a comment to be explicit about this.